SpatialGDK/Source/SpatialGDK/Public/Tests/SpatialView/TestWorker.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "SpatialView/ViewCoordinator.h"
+#include "TestConnectionHandler.h"
+
+namespace SpatialGDK
+{
+class FTestWorker
+{
+public:
+	explicit FTestWorker(TArray<Worker_ComponentSetId> ComponentSetIds, FSendMessageCallback InSendMessageCallback = FSendMessageCallback(),
+						 FString InWorkerId = {}, Worker_EntityId InWorkerSystemEntityId = 0)
+	{
+		TUniquePtr<FTestConnectionHandler> Handler =
+			MakeUnique<FTestConnectionHandler>(InSendMessageCallback, InWorkerId, InWorkerSystemEntityId);
+		ConnectionHandler = Handler.Get();
+		FComponentSetData SetData;
+		for (Worker_ComponentSetId SetId : ComponentSetIds)
+		{
+			SetData.ComponentSets.Add(SetId, {});
+		}
+		Coordinator = ViewCoordinator(MoveTemp(Handler), nullptr, MoveTemp(SetData));
+	}
+
+	void AddOpList(OpList Ops) { ConnectionHandler->AddOpList(MoveTemp(Ops)); }
+
+	ViewCoordinator& GetCoordinator() { return Coordinator; }
+
+	const ViewCoordinator& GetCoordinator() const { return Coordinator; }
+
+private:
+	FTestConnectionHandler* ConnectionHandler;
```

Comment:
See comment above - should this not be a TUniquePtr?

Reply:
It can't be a unique pointer as the connection is owned by the coordinator not this wrapper. We could make it a shared pointer although that also feels strange. I think it probably shouldn't be a member and instead be a thing that take from the coordinator when we need it and upcast it.